themes/default/content/blog/pulumi-security-posture/index.md

```diff
@@ -0,0 +1,97 @@
+
+Risks associated with maintaining a good Cloud Security Posture extend to your Infrastructure as Code with Pulumi and it’s important to leverage the features and capabilities of the Pulumi Platform along with good industry practices to secure your code, configuration and secrets.
+
+## Pulumi’s Hierarchy
+
+Pulumi has three levels to consider access for: Users, Teams and Organizations.
+
+[Users](https://www.pulumi.com/docs/intro/pulumi-service/accounts/) are tied to personal identities often with an Identity Provider and should be used wherever a human is using Pulumi.  They should not be used for any systems, tools or pipeline based provisioning.
+
+[Teams](https://www.pulumi.com/docs/intro/pulumi-service/teams/) are groups of Users within an Organization with specific permissions to Stacks.  Consider using these to break up access to environments such as Dev, Staging and Production.  You should look to mirror your code repository or IAM teams/groups as much as possible.
+
+[Organizations](https://www.pulumi.com/docs/intro/pulumi-service/organizations/) are a space for you to collaborate on shared projects and stacks.  You may have multiple Cloud Accounts from a single cloud provider or multiple cloud provider Accounts tied to an Organization in any combination.  It’s recommended that you segment Organizations and Cloud Accounts to limit access and the “blast radius” of security or financial events that may occur within them.
+
+## Security Best Practices with Pulumi
+
+The following recommendations should help you get started and prepare you to scale your company, application and cloud engineering team long term.
+
+## Manage Federated Identity
+
+Connect your user and Pulumi Orgs with an [Identity Provider](https://www.pulumi.com/docs/intro/pulumi-service/organizations/#organization-identity-providers) such as [Github](https://www.pulumi.com/docs/intro/pulumi-service/organizations/#github-identity-provider), [Gitlab](https://www.pulumi.com/docs/intro/pulumi-service/organizations/#gitlab-identity-provider) or [BitBucket](https://www.pulumi.com/docs/intro/pulumi-service/organizations/#bitbucket-identity-provider), your [Single Sign On (SSO)](https://www.pulumi.com/docs/intro/pulumi-service/organizations/#saml-single-sign-on-sso) system and/or another [System for Cross-Domain Identity Management](https://www.pulumi.com/docs/guides/scim/) (SCIM).  Your code repositories hold your Infrastructure as Code crown jewels, and most organizations are motivated to keep access tightly controlled.  SSO improves this process across tools and puts more control into your organization’s hands.  SCIM enables you to manage your users and groups centrally in your Identity Provider (IdP) and then synchronize those users and groups to the Pulumi Service.  Leveraging your existing onboarding and offboarding process will help limit the risk of bad actors in Pulumi itself.
```

Comment:
```suggestion
Connect your user and Pulumi orgs with an [Identity Provider](https://www.pulumi.com/docs/intro/pulumi-service/organizations/#organization-identity-providers) such as [Github](https://www.pulumi.com/docs/intro/pulumi-service/organizations/#github-identity-provider), [Gitlab](https://www.pulumi.com/docs/intro/pulumi-service/organizations/#gitlab-identity-provider) or [BitBucket](https://www.pulumi.com/docs/intro/pulumi-service/organizations/#bitbucket-identity-provider), your [Single Sign On (SSO)](https://www.pulumi.com/docs/intro/pulumi-service/organizations/#saml-single-sign-on-sso) system and/or another [System for Cross-Domain Identity Management](https://www.pulumi.com/docs/guides/scim/) (SCIM).  Your code repositories hold your Infrastructure as Code crown jewels, and most organizations are motivated to keep access tightly controlled.  SSO improves this process across tools and puts more control into your organization’s hands.  SCIM enables you to manage your users and groups centrally in your Identity Provider (IdP) and then synchronize those users and groups to the Pulumi Service.  Leveraging your existing onboarding and offboarding process will help limit the risk of bad actors in Pulumi itself.
```

Reply:
Could use a little more explanation of what you mean by "crown jewels" since these repos should definitely not contain secrets.

Reply:
Removed `crown jewels` wording